src/main/java/gregtech/api/recipes/RecipeMap.java

```diff
@@ -357,7 +365,7 @@ public static ItemStack[] uniqueItems(Collection<ItemStack> input) {
      */
 
     public static List<GTRecipeInput> uniqueIngredientsList(List<GTRecipeInput> input) {
-        List<GTRecipeInput> list = new ObjectArrayList<>(input.size());
+        LinkedList<GTRecipeInput> list = new LinkedList<>();
```

Comment:
Does `ObjectArrayList#add(int index, E entry)` work here also instead of `LinkedList#addFirst(E entry)`?

It may also be possible to give each type of ingredient a priority index, and do the sort at the very end.